From 303b65a754d8cf2df34396757f30b49d881cae96 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:44:38 -0400
Subject: [x86] pci: Use pcibios_scan_root() instead of pci_scan_bus_on_node()

Message-id: <20140806034437.1438.95379.stgit@gir.stowe>
Patchwork-id: 87414
O-Subject: [RHEL7.1 PATCH 05/69] x86/PCI: Use pcibios_scan_root() instead of pci_scan_bus_on_node()
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 3d2a366190778f30a2f10b5edaf32bcb236840fd

commit 3d2a366190778f30a2f10b5edaf32bcb236840fd
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Fri Jan 24 11:50:48 2014 -0700

    x86/PCI: Use pcibios_scan_root() instead of pci_scan_bus_on_node()

    pcibios_scan_root() looks up the bus's NUMA node, then calls
    pci_scan_bus_on_node().  This uses pcibios_scan_root() directly and drops
    the node lookup in the callers.

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Acked-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/pci/irq.c b/arch/x86/pci/irq.c
index 372e9b8..8658874 100644
--- a/arch/x86/pci/irq.c
+++ b/arch/x86/pci/irq.c
@@ -136,11 +136,9 @@ static void __init pirq_peer_trick(void)
   busmap[e->bus] = 1;
  }
  for (i = 1; i < 256; i++) {
-  int node;
   if (!busmap[i] || pci_find_bus(0, i))
    continue;
-  node = get_mp_bus_to_node(i);
-  if (pci_scan_bus_on_node(i, &pci_root_ops, node))
+  if (pcibios_scan_root(i))
    printk(KERN_INFO "PCI: Discovered primary peer "
           "bus %02x [IRQ]\n", i);
  }
diff --git a/arch/x86/pci/legacy.c b/arch/x86/pci/legacy.c
index 4db96fb..5b662c0 100644
--- a/arch/x86/pci/legacy.c
+++ b/arch/x86/pci/legacy.c
@@ -37,19 +37,17 @@ int __init pci_legacy_init(void)
 void pcibios_scan_specific_bus(int busn)
 {
  int devfn;
- long node;
  u32 l;
 
  if (pci_find_bus(0, busn))
   return;
 
- node = get_mp_bus_to_node(busn);
  for (devfn = 0; devfn < 256; devfn += 8) {
   if (!raw_pci_read(0, busn, devfn, PCI_VENDOR_ID, 2, &l) &&
       l != 0x0000 && l != 0xffff) {
    DBG("Found device at %02x:%02x [%04x]\n", busn, devfn, l);
    printk(KERN_INFO "PCI: Discovered peer bus %02x\n", busn);
-   pci_scan_bus_on_node(busn, &pci_root_ops, node);
+   pcibios_scan_root(busn);
    return;
   }
  }
-- 
1.7.1